packages/main/src/plugin/docker/docker-compatibility.ts

```diff
@@ -46,4 +60,106 @@ export class DockerCompatibility {
 
     this.#configurationRegistry.registerConfigurations([dockerCompatibilityConfiguration]);
   }
+
+  protected getTypeFromServerInfo(info: { OperatingSystem?: string }, podmanInfo: unknown): DockerSocketServerInfoType {
+    if (info.OperatingSystem === 'Docker Desktop') {
+      return 'docker';
+    } else if (info.OperatingSystem === 'podman' || podmanInfo) {
+      // if podman info is available, then it is podman
+      return 'podman';
+    }
+    return 'unknown';
+  }
+
+  async resolveLinkIfAny(path: string): Promise<string> {
+    // do not resolve on Windows
+    if (isWindows()) {
+      return path;
+    }
+
+    const lstat = await promises.lstat(path);
+    if (lstat.isSymbolicLink()) {
+      const resolved = await promises.readlink(path);
+      return this.resolveLinkIfAny(resolved);
+    }
+
+    return path;
+  }
+
+  async getSystemDockerSocketMappingStatus(): Promise<DockerSocketMappingStatusInfo> {
+    // first, try to connect to the docker socket if on Linux or macOS
+    // if the connection is successful, return @the status as running
```

Comment:
Is this comment still accurate?

Reply:
accurate but I'll change the wording as it looks like it's unclear 👍

Reply:
fixed, please review